tests/qref/data/invalid_topology_programs.yaml

```diff
@@ -0,0 +1,136 @@
+- input:
+    program:
+      children:
+      - name: child_1
+        ports:
+        - direction: input
+          name: in_0
+          size: 1
+        - direction: output
+          name: out_0
+          size: 1
+        - direction: input
+          name: in_1
+          size: 1
+        - direction: output
+          name: out_1
+          size: 1
+      - name: child_2
+        ports:
+        - direction: input
+          name: in_0
+          size: 1
+        - direction: output
+          name: out_0
+          size: 1
+        - direction: input
+          name: in_1
+          size: 1
+        - direction: output
+          name: out_1
+          size: 1
+      - name: child_3
+        ports:
+        - direction: input
+          name: in_0
+          size: 1
+        - direction: output
+          name: out_0
+          size: 1
+        - direction: input
+          name: in_1
+          size: 1
+        - direction: output
+          name: out_1
+          size: 1
+
+      connections:
+      - source: in_0
+        target: child_1.in_0
+      - source: child_1.out_0
+        target: child_2.in_0
+      - source: child_2.out_0
+        target: child_3.in_0
+      - source: child_3.out_0
+        target: out_0
+      - source: child_1.out_1
+        target: child_2.in_1
+      - source: child_2.out_1
+        target: child_3.in_1
+      - source: child_3.out_1
+        target: child_1.in_1
+
+      name: root
+      ports:
+      - direction: input
+        name: in_0
+        size: 1
+      - direction: output
+        name: out_0
+        size: 1
+    version: v1
+  description: Program contains cycles
+  problems:
+  - "Cycle detected for node: *"
```

Comment:
**suggestion**: Nice examples, I would only add what cycle we do expect to find here, otherwise the reader has to go through the connections manually (and the fact that we use flat graph makes it even harder!).

Reply:
Ok, done.